common/containers/Tabs/Swap/components/wantToSwapMy.js

```diff
@@ -0,0 +1,143 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import translate from 'translations';
+import {combineAndUpper} from 'api/bity';
+
+function sleep(ms) {
+    return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+export default class WantToSwapMy extends Component {
+    constructor(props) {
+        super(props)
+    }
+
+    static propTypes = {
+        bityRates: PropTypes.any,
+        originAmount: PropTypes.any,
+        destinationAmount: PropTypes.any,
+        originKind: PropTypes.string,
+        destinationKind: PropTypes.string,
+        destinationKindOptions: PropTypes.array,
+        originKindOptions: PropTypes.array,
+        swapOriginKindTo: PropTypes.func,
+        swapDestinationKindTo: PropTypes.func,
+        swapOriginAmountTo: PropTypes.func,
+        swapDestinationAmountTo: PropTypes.func
+    };
+
+    onClickStartSwap() {
+
+    }
+
+    onChangeOriginAmount = (amount) => {
+        let originAmountAsNumber = parseFloat(amount);
+        if (originAmountAsNumber) {
+            let pairName = combineAndUpper(this.props.originKind, this.props.destinationKind);
+            let bityRate = this.props.bityRates[pairName];
+            this.props.swapOriginAmountTo(originAmountAsNumber);
+            this.props.swapDestinationAmountTo(originAmountAsNumber * bityRate)
+        } else {
+            this.props.swapOriginAmountTo('');
+            this.props.swapDestinationAmountTo('')
+        }
+    };
+
+    onChangeDestinationAmount(amount) {
+        let destinationAmountAsNumber = parseFloat(amount);
+        if (destinationAmountAsNumber) {
+            this.props.swapDestinationAmountTo(destinationAmountAsNumber);
+            let pairName = combineAndUpper(this.props.destinationKind, this.props.originKind);
+            let bityRate = this.props.bityRates[pairName];
+            this.props.swapOriginAmountTo(destinationAmountAsNumber * bityRate)
+        } else {
+            this.props.swapOriginAmountTo('');
+            this.props.swapDestinationAmountTo('')
+        }
+    }
+
+    async onChangeDestinationKind(event) {
+        let toKind = event.target.value;
+        this.props.swapDestinationKindTo(toKind);
+        // TODO - can't find a way around this without bringing in annoying deps. Even though redux action is sync,
+        // it seems it happens in the background and values don't get updated in time
+        await sleep(100);
```

Comment:
you just need 1 tick, no reason to sleep any longer. Also this should tell us we need to hoist more state and logic to redux.